lib/src/configuration.dart

```diff
@@ -405,3 +411,19 @@ class SyncClientResetErrorHandler {
 
 /// A client reset strategy where the user needs to fully take care of a client reset.
 typedef ManualSyncClientResetHandler = SyncClientResetErrorHandler;
+
+/// Represents an object with [FlexibleSyncConfiguration] configuration options for adding subscriptions initially before a [Realm] is opened.
+///
+/// {@category Configuration}
+class InitialSubscriptionsConfiguration {
+
+  const InitialSubscriptionsConfiguration(this.callback, {this.rerunOnOpen = false});
+
+  /// The callback that will be called in order to populate the initial [SubscriptionSet] for the [Realm].
+  final void Function(Realm) callback;
+
+  /// The default behavior is that callback is only invoked the first time the Realm is opened,
+  /// but if rerunOnOpen is true, it will be invoked every time the realm is opened.
+  final bool rerunOnOpen;
```

Comment:
I don't understand. We don't open the app.

Reply:
The idea is that the callback is invoked every time the app is launched, not every time the Realm file is opened. These may be the same if you only open the Realm once for the lifetime of your app, but if you open the Realm on every screen, it will be a problem.

Reply:
Added a discussion point.

Reply:
Added core issue: https://github.com/realm/realm-core/issues/5962

Reply:
I disagree very very strongly on the idea that we should make rerunOnOpen literally once per run of an app.

As sort of a broad thing, we should be trying to have *less* implicit global state in core, not more. What we have already has always been a problem for things like tests and platforms where multiple VM states per process are normal, and we shouldn't be making the problem worse.

More specific to this, it's not an accident or bug that things like compactOnLaunch and the next-launch file actions for sync are actually "when the Realm is opened from scratch" and not once per run of the app. We've implied that they're once per app launch because we expect that to be how they normally work out (especially with synchronized Realms), but performing some operation a second time should never require killing an app and relaunching it.

Opening the Realm on each screen will typically not actually result in the Realm being reopened from scratch each time as the SyncSession isn't torn down synchronously. If the app does manage to actually fully close the Realm and reopen it on each screen transition then performance will already be very poor.